libraries/AP_Common/Location.cpp

```diff
@@ -194,7 +210,7 @@ bool Location::get_vector_xy_from_origin_NE(Vector2f &vec_ne) const
         return false;
     }
     vec_ne.x = (lat-ekf_origin.lat) * LATLON_TO_CM;
-    vec_ne.y = diff_longitude(lng,ekf_origin.lng) * LATLON_TO_CM * ekf_origin.longitude_scale();
+    vec_ne.y = diff_longitude(lng,ekf_origin.lng) * LATLON_TO_CM * longitude_scale((lat+ekf_origin.lat)/2);
```

Comment:
this should help improve E/W accuracy at larger distances from the origin

Reply:
All uses of longitude_scale that I could find are now using the average latitude for the two points so this change should not result in any inconsistencies at larger distances from the origin.

Reply:
yes, it preserves the basic (a+b)-b == a rule, which I think is fairly fundamental